arm_robots/src/arm_robots/base_robot.py

```diff
@@ -96,3 +110,34 @@ def check_inputs(self, group_name: str, ee_link_name: str):
         if group_name not in groups:
             rospy.logwarn_throttle(1, f"Group [{group_name}] does not exist. Existing groups are:")
             rospy.logwarn_throttle(1, groups)
+
+    def create_cartesian_impedance_controller(self, motion_status_listeners: List[Listener],
+                                              motion_command_publishers: List[rospy.Publisher],
+                                              joint_names: List[str],
+                                              world_frame_name: str,
+                                              **kwargs):
+        """
+        Create the cartesian impedance controller. The number of listeners should match the number of publishers and
+        each element corresponds to 1 arm. We assume all arms share the same joint properties, so only 1 list of joint
+        names should be supplied.
+        """
+        lower, upper = self.get_joint_limits(joint_names, safety_margin=0)
+        self.cartesian = CartesianImpedanceController(self.tf_wrapper.tf_buffer,
+                                                      motion_status_listeners,
+                                                      motion_command_publishers,
+                                                      lower, upper, world_frame_name, **kwargs)
+
+    def move_delta_cartesian_impedance(self, arm, dx, dy, target_z=None, target_orientation=None,
+                                       step_size=0.01, blocking=True):
+        if self.cartesian is None:
+            return False
+
+        self.cartesian.set_active_arm(arm)
+        if not self.cartesian.set_relative_goal_2d(dx, dy, target_z=target_z, target_orientation=target_orientation):
+            return False
+        succeeded = self.cartesian.step(step_size)
```

Comment:
this is kind of weird. why step once, and why return the value of this single step? I would expect perhaps `return None` in the non-blocking case?

Reply:
Yeah I wasn't sure what the ideal API for allowing blocking/non-blocking calls is. The idea behind stepping once and returning its success is to check against invalid starting configurations. Not stepping and just returning `True` is probably cleaner. The return would be whether the set relative goal succeeded. Returning None would be confusing since they'd have to check the output against {None, True, False}, and if they just check truthiness then the None would be implicitly converted to False.